src/engine/strat_engine/filesystem.rs

```diff
@@ -94,20 +96,22 @@ impl StratFilesystem {
 
     /// Get the mount_point for this filesystem
     /// TODO Replace this code with something less brittle
-    pub fn get_mount_point(&self) -> EngineResult<PathBuf> {
+    pub fn get_mount_point(&self) -> EngineResult<Option<PathBuf>> {
         let output = Command::new("df")
             .arg("--output=target")
             .arg(&self.devnode())
             .output()?;
         if output.status.success() {
             let output_str = String::from_utf8_lossy(&output.stdout);
             if let Some(mount_point) = output_str.lines().last() {
-                return Ok(PathBuf::from(mount_point));
+                return Ok(Some(PathBuf::from(mount_point)));
```

Comment:
Whole block might as well be:

```return Ok(output_str.lines().last().map(|mp| PathBuf::from(mp)));```

Reply:
Ok

Reply:
Clippy complains about the suggested code:

155 |             return Ok(output_str.lines().last().map(|mp| PathBuf::from(mp)));

    |                                                     ^^^^^^^^^^^^^^^^^^^^^^ help: remove closure as shown: `PathBuf::from`

I'll switch back to the original.

Reply:
What clippy means is that the line is too ornate. Do the following, instead:

```
return Ok(output_str.lines().last().map(PathBuf::from));
```

Reply:
Ok - this works.